src/Umbraco.Infrastructure/Persistence/Repositories/Implement/DistributedJobRepository.cs

```diff
@@ -0,0 +1,124 @@
+﻿using NPoco;
+using Umbraco.Cms.Core.Exceptions;
+using Umbraco.Cms.Infrastructure.BackgroundJobs;
+using Umbraco.Cms.Infrastructure.Models;
+using Umbraco.Cms.Infrastructure.Persistence.Dtos;
+using Umbraco.Cms.Infrastructure.Scoping;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement;
+
+/// <inheritdoc />
+internal class DistributedJobRepository(IScopeAccessor scopeAccessor) : IDistributedJobRepository
+{
+    /// <inheritdoc />
+    public string? GetRunnable()
+    {
+        if (scopeAccessor.AmbientScope is null)
+        {
+            return null;
+        }
+
+        Sql<ISqlContext> sql = scopeAccessor.AmbientScope.SqlContext.Sql()
+            .Select<DistributedJobDto>()
+            .From<DistributedJobDto>()
+            .Where<DistributedJobDto>(x => x.IsRunning == false)
+            .OrderBy<DistributedJobDto>(x => x.LastRun);
+
+        IUmbracoDatabase database = scopeAccessor.AmbientScope.Database;
+        List<DistributedJobDto> jobs = database.Fetch<DistributedJobDto>(sql);
+
+        // Grab the first runnable job.
+        DistributedJobDto? job = jobs.FirstOrDefault(x => x.LastRun < DateTime.UtcNow - TimeSpan.FromTicks(x.Period));
```

Comment:
This is a lot of business logic for a repository. This should be in the service instead, especially since I don't see what we gain from this being in the repository. So we'd 

So in the service, we should instead
1. Fetch all jobs from the repository
2. Then find the runnable job
3. Then return null if there is none
4. If there is one, then update the database through the service

Otherwise, what constitutes a "runnable job" is determined by the repository,y which should only concern itself with getting and storing data IMO. 

The logic itself is sound though :smile:

Reply:
Good point, fixed 💪